src/snowflake/snowpark/mock/_snowflake_data_type.py

```diff
@@ -1,7 +1,15 @@
 #
 # Copyright (c) 2012-2024 Snowflake Computing Inc. All rights reserved.
 #
-from typing import Dict, NamedTuple, Optional, Union
+from typing import Any, Dict, Iterable, NamedTuple, Optional, Union
+
+from pandas.core.dtypes.common import (
+    is_bool_dtype,
+    is_float_dtype,
+    is_integer_dtype,
+    is_object_dtype,
+    is_string_dtype,
+)
 
```

Comment:
This import will need to be conditional because pandas is optional dependency.

Reply:
Addressed in https://github.com/snowflakedb/snowpark-python/pull/2512/files. Not 100% clear here on the relationship of local testing with pandas. I thought localtesting was built with pandas, so pandas is a dependency? I assume you're referring here to some potential import problems. Fixed by making this imports local imports and adding a check for `if not installed_pandas: raise ...`.